Move out-of-order packets definition (#3369)
quicwg/base-drafts: draft-ietf-quic-transport.md

  CONNECTION_CLOSE. These cause a recipient to send an acknowledgment (see
  {{sending-acknowledgements}}).

> Out-of-order Packets:

Reviewer: ```suggestion
Out-of-order packet:
```

> : Packets that do not increase the largest received packet number for its

Reviewer: ```suggestion
: A packet that does not increase the largest received packet number for its
```

>   packet number space by exactly one. Packets arrive out of order

Reviewer: ```suggestion
  packet number space by exactly one. A packet can arrive out of order
```

>   when earlier packets are lost or delayed.

Reviewer 2: Do we need to point out that an endpoint also might skip a packet number intentionally, as suggested in [section 21.3](https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#name-optimistic-ack-attack)?

Or maybe we should state in that section that skipping packet numbers too frequently would lead to an unnecessarily large number of ACK packets being sent in response.

Not that I have a strong opinion here, but I just wanted to make sure we'd be fine with what we'd have.

Reviewer: I think it's worth mentioning something in 21.3, but I don't think it belongs in this definition.

>   when earlier packets are lost or delayed.

Reviewer: ```suggestion
  if it is delayed or if earlier packets are lost or delayed.
```

> 

Reviewer 3: Technically, a packet can also arrive out of order if a packet number is intentionally skipped, typically for optimistic ACK attack protection.

Author: We're using it to elicit an ACK more quickly in the case of a single PTO packet, see #2952
